cmd/kritis/kubectl/plugins/resolve/main.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2018 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"gopkg.in/yaml.v2"
+	"os"
+)
+
+var (
+	files = os.Args[1:]
+)
+
+func main() {
+	if err := execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func execute() error {
+	// TODO (priyawadhwa): fill in this function
+	//	1. Resolve relative paths in files to absolute paths
+	//  2. Get tagged images using recursiveGetTaggedImages
+	// 	3. Resolve tagged images using resolveTagsToDigests
+	//	4. Write recursive function to replace images
+	//	5. Print to STDOUT
+	return nil
+}
+
+// recursiveGetTaggedImages recursively gets all images referenced by tags
+// instead of digests
+func recursiveGetTaggedImages(m interface{}) []string {
```

Comment:
This should work, but it's worth considering doing this the same way kubectl does, which is essentially a more typed version of the same thing: https://github.com/kubernetes/kubernetes/blob/9eb0c3566808bb02de5ae6f43fa79da49718c864/pkg/kubectl/polymorphichelpers/updatepodspec.go#L33

Reply:
So I don't think this can be further typed beyond `yaml.MapItem`, `yaml.MapSlice`, and `[]interface{}` (which is meant to represent `[]yaml.MapSlice`), unless I'm missing something?